sdk/src/main/java/com/meniga/sdk/models/budget/MenigaBudget.java

```diff
@@ -187,12 +196,41 @@ public MenigaBudget createFromParcel(Parcel source) {
 		return apiOperator.getBudget(FetchBudgetFilterExtensions.toGetBudget(filter));
 	}
 
-	public static Result<MenigaBudget> create(BudgetType type, String name, String description, List<Long> accountIds, BudgetPeriod period) {
-		return apiOperator.createBudget(type, name, description, accountIds, period, null);
+	/**
+	 * Use {@link #create(NewBudget)} instead.
+	 */
+	@Deprecated
+	public static Result<MenigaBudget> create(
+			@Nonnull BudgetType type,
+			@Nonnull String name,
+			@Nullable String description,
+			@Nullable List<Long> accountIds,
+			@Nullable BudgetPeriod period) {
+		return create(type, name, description, accountIds, period, null);
 	}
 
-	public static Result<MenigaBudget> create(BudgetType type, String name, String description, List<Long> accountIds, BudgetPeriod period, Integer periodOffset) {
-		return apiOperator.createBudget(type, name, description, accountIds, period, periodOffset);
+	/**
+	 * Use {@link #create(NewBudget)} instead.
+	 */
+	@Deprecated
+	public static Result<MenigaBudget> create(
+			@Nonnull BudgetType type,
+			@Nonnull String name,
+			@Nullable String description,
+			@Nullable List<Long> accountIds,
+			@Nullable BudgetPeriod period,
```

Comment:
At least for BudgetType.PLANNING, budget period and period offset can't both be null at the same time.

Reply:
You're right. The API will respond with an error if we set it to null for `BudgetType.PLANNING` but it's completely OK for `BudgetType.BUDGET`.

I would keep it like that because the user will receive an error response when doing that. If we want to simplify it for our users and add this logic to the SDK we can split it to separate methods:

```
create(NewPlanningBudget)
create(NewBudget)
```

in which case we would check if the data is valid on our side. What do you think about it @agustk ?

Reply:
I think that's a good suggestion and would make our SDK clearer to the user. Can you make it so?

Reply:
Done